mipengine/algorithms/logistic_regression.py

```diff
@@ -7,223 +7,245 @@
 import numpy
 import pandas as pd
 
-from mipengine.algorithms import udf
-from mipengine.algorithms import TableT
-from mipengine.algorithms import TensorT
-from mipengine.algorithms import LiteralParameterT
-from mipengine.algorithms import ScalarT
-from mipengine.algorithms import RelationT
+from mipengine.node.udfgen import (
+    TensorBinaryOp,
+    TensorUnaryOp,
+    literal,
+    make_unique_func_name,
+    merge_tensor,
+    relation,
+    scalar,
+    tensor,
+    udf,
+)
 
 
 PREC = 1e-6
 
 
 def run(algo_interface):
-
-    # y: TableT, X: TableT
-    run_on_locals = algo_interface.run_udf_on_local_nodes
-    run_on_global = algo_interface.run_udf_on_global_node
+    local_run = algo_interface.run_udf_on_local_nodes
+    global_run = algo_interface.run_udf_on_global_node
     get_table_schema = algo_interface.get_table_schema
 
-    X_init: "LocalNodeTable" = algo_interface.initial_view_tables["x"]
-    X = run_on_locals(
-        func_name="logistic_regression.relation_to_matrix",
-        positional_args={"rel": X_init},
-    )  # X = relation_to_matrix(X)
+    X_relation: "LocalNodeTable" = algo_interface.initial_view_tables["x"]
+    X = local_run(
+        func_name=make_unique_func_name(relation_to_matrix),
```

Comment:
After a call we decided to make a jira task, so that `make_unique_func_name` will be inside the algorithm executor in the future.